detectable/src/main/java/com/blackduck/integration/detectable/detectables/gradle/inspection/parse/GradleReportLineParser.java

```diff
@@ -217,10 +220,14 @@ private void checkParentRichVersion(String dependencyGroupName, String currentPr
         // this loop checks all the parent modules for the current submodule upto rootProject for the use of the rich version for the current dependency
         // if the rich version is used return true and update the richVersionProject
         // this loop will stop at the root project and traverse all the parents upto root project, the change was done to support all projects which use a project structure where two child submodules with same name
-        if (gradleRichVersions.containsKey(currentProject) && gradleRichVersions.get(currentProject).containsKey(dependencyGroupName)) {
-            richVersionProject = currentProject;
-            richVersionUsed = true;
+        // In order to tight security for the same name child submodules and the parent who declares rich version, we make sure if the parent project we are checking comes in project path of current project
+        //Eg: if sub declares rich version and foo is child of both sub and subtwo, we change version if :sub:foo is the path we are parsing and do not change if we are parsing :subtwo:foo
+        if (gradleRichVersions.containsKey(currentProject) && gradleRichVersions.get(currentProject).containsKey(dependencyGroupName) && (currentProject.equals(rootProjectName) || projectPath.contains(currentProject.split("_")[0]))) {
```

Comment:
For the first question, yes we are trying to check if rootProject has declared the rich version and it would not collide with subproject name because rootProjectName would have depth 0 which can only be one in entire project. 

For the second question, separator refers to the project name that we define in the key i.e name_level, so it can refer to one thing only. Nope it does not matter, we are just checking if the parent project that we are checking is in full path of the child submodule.

P.S: I am trying to eliminate this relationsMap all over and just use projectPath to figure out if a parent module have declared a rich version.

Reply:
Updated the approach to remove use of that complex flow and simply use projectPath to loop over parent modules.